docs/groovy/reference/community-questions/shifts.md

```diff
@@ -0,0 +1,129 @@
+---
+title: How do row keys and positional indices behave during merge table operations?
+sidebar_label: How do row keys and positional indices behave during merge table operations?
+---
+
+_When merging tables in Deephaven, I understand that "shifts" can happen when one table grows beyond its allocated row key slots. How do the row key (k) and positional index (i) attributes change during these shifts, and what impact does this have on downstream operations?_
+
+When you merge tables in Deephaven, the engine pre-allocates row key slots for each constituent table. Understanding how row keys (`k`) and positional indices (`i`) behave during growth and shifts is crucial for working with formulas that reference these attributes.
+
+> [!DANGER]
+> **Formulas using positional indices (`i`, `ii`, `k`) or column array variables are unsafe on refreshing tables and will cause `IllegalArgumentException` errors.** The engine blocks these formulas by default to prevent incorrect results. While you can override this with the system property `io.deephaven.engine.table.impl.select.AbstractFormulaColumn.allowUnsafeRefreshingFormulas=true`, doing so **will produce incorrect results** when table shifts occur. **Note that shifts can happen with any table operation, not just merges** - including joins, updates, selects, and other operations on live/refreshing tables. Only use such formulas on static tables or append-only tables where shifts cannot happen.
```

Comment:
This is rendered weird. It doesn't look like other danger/caution/warning/notes.

Additionally, as Charles pointed out, these variables work on append-only tables (hence why you can do `time_table("PT1s").update("X = ii")`.